code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -738,3 +738,54 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 		var/list/color = rgb2num(owner.material.getColor())
 		light_c = owner.AddComponent(/datum/component/loctargeting/sm_light, color[1], color[2], color[3], 255 * 0.33)
 		light_c.update(1)
+
+/datum/materialProc/radioactive_temp
+	max_generations = -1
+
+	execute(var/atom/owner, var/temp)
+		if (isobj(owner) && temp > 500 KELVIN)
```

Comment:
Nitpick but these kind of conditions are better as guards, ie `if (!condition) return`, leads to less indent chaos